### PR TITLE
Take AsRef<Path> arguments for path names, not str

### PR DESCRIPTION
**Pull request by asomers**

### CODE REVIEW

**Review by dlrobertson — APPROVED** [review]
This is great! Thanks!

File: src/util.rs [hunk 1]
```diff
@@ -35,26 +37,26 @@ use common::{CapResult, CapErr, CapErrType};
...
+    pub fn new<P: AsRef<Path>>(path: P) -> io::Result<Directory> {
+        let file = File::open(path.as_ref())?;
         Ok(Directory {
             file,
         })
     }
 
-    pub fn open_file(&self, path: CString, flags: c_int, mode: Option<mode_t>) -> CapResult<File> {
+    pub fn open_file<P: AsRef<Path> + ?Sized>(&self, path: &P, flags: c_int, mode: Option<mode_t>) -> CapResult<File> {
+        let p = path.as_ref().as_os_str().as_bytes().as_ptr() as *const i8;
```
Comment (dlrobertson) [inline src/util.rs:55]: I think this should be something like:

```
let path_str = CString::new(path.as_ref().as_os_str().as_bytes()).map_err(<something>)?;
...
    openat(self.file.as_raw_fd(), path_str.as_ptr(), ...)
```

**Review by dlrobertson — COMMENTED** [review]
Not entirely sure why the tests started failing after a rebase, but my best guess is the string conversion for the directory path.

File: src/util.rs (hunk 1 above)
Comment (asomers) [reply src/util.rs:55]: Yep, that was it.

**Pull request merged**